disnake/poll.py

```diff
@@ -0,0 +1,426 @@
+
+
+class PollMedia:
+    """Represents a poll media from discord.
+
+    Parameters
+    ----------
+    text: Optional[:class:`str`]
+        The text of this media.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji of this media.
+
+
+    .. note::
+
+        You are expected to pass at least text or emoji. They can't be both none at the same time.
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of this media.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji of this media.
+    """
+
+    __slots__ = ("text", "emoji")
+
+    def __init__(
+        self, text: Optional[str] = None, *, emoji: Optional[Union[Emoji, PartialEmoji, str]] = None
+    ) -> None:
+        if text is None and emoji is None:
+            raise ValueError("Either one of text or emoji must be not none.")
+
+        self.text = text
+        self.emoji = emoji
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} text={self.text} emoji={self.emoji!r}>"
+
+    @classmethod
+    def from_payload(cls, data: PollMediaPayload) -> PollMedia:
+        text = data.get("text")
+
+        emoji = None
+        if _emoji := data.get("emoji"):
+            emoji = PartialEmoji.from_dict(_emoji)
+
+        return cls(text=text, emoji=emoji)
+
+    def _to_dict(self) -> PollCreateMediaPayload:
+        payload: PollCreateMediaPayload = {}
+        if self.text:
+            payload["text"] = self.text
+        if self.emoji:
+            if isinstance(self.emoji, (Emoji, PartialEmoji)):
+                if self.emoji.id:
+                    payload["emoji"] = {"id": self.emoji.id}
+                else:
+                    payload["emoji"] = {"name": self.emoji.name}
+            elif isinstance(self.emoji, str):
+                payload["emoji"] = {"name": self.emoji}
+        return payload
+
+
+class PollAnswer:
+    """Represents a poll answer from discord.
+
+    Parameters
+    ----------
+    text: Optional[:class:`str`]
+        The text of the answer.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji of the answer.
+
+        .. note::
+
+            If this is str you are expected to pass the name of the emoji. This works only for discord default emojis. For custom emojis you need to use an Emoji or PartialEmoji object.
+
+    Attributes
+    ----------
+    answer_id: :class:`int`
+        The ID of the answer that this object represents.
+    poll_media: :class:`PollMedia`
+        The media fields linked to this answer.
+    """
+
+    __slots__ = ("_state", "_channel_id", "_message_id", "answer_id", "poll_media")
+
+    def __init__(
+        self, text: Optional[str] = None, *, emoji: Optional[Union[Emoji, PartialEmoji, str]] = None
+    ) -> None:
+        self._state: ConnectionState
+        self._channel_id: int
+        self._message_id: int
+        self.answer_id: int
```

Comment:
Besides the undefined attributes issue - presumably there's always a corresponding `Message` object whenever we have a `Poll` or `PollAnswer` object? So I don't really see the use in storing multiple ID fields instead of a `self._message: Optional[Message] = None`, from which you can retrieve both the channel and state when needed